### PR TITLE
Support resource configuration for imagebuilder

### PR DESCRIPTION
**Pull request by terryyylim**
**What this PR does / why we need it**:

Currently, Merlin image building jobs don't support resource configuration, and requests.cpu is hardcoded. Without requests.memory configured on them, it's common to see jobs failing with OOMKilled error if multiple jobs are running concurrently. This PR supports resource configuration so that some reasonable amount of memory can be set to mitigate this issue. In addition, the GKE autoscaler can spin up another node if multiple jobs are running concurrently, instead of placing all of the jobs onto the same node, and causing them to fight for available memory.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

- [x] Added unit test, integration, and/or e2e tests

### CODE REVIEW

**Comment by pradithya** [comment]
@terryyylim LGTM, can you update the helm chart as well?

**Review by pradithya — APPROVED** [review]

File: charts/merlin/values-e2e.yaml [hunk 1]
```diff
@@ -74,6 +74,13 @@ merlin:
         dockerfilePath: "docker/app.Dockerfile"
         buildContextURI: "git://github.com/gojek/merlin.git#refs/tags/v0.19.0"
         mainAppPath: /merlin-spark-app/main.py
+    resourceRequestsLimits:
```
Comment (pradithya) [inline charts/merlin/values-e2e.yaml:77]: nit: I think `resources` is more concise.

**Review by pradithya — APPROVED** [review]
One little nitpick, LGTM thanks @terryyylim

**Comment by ariefrahmansyah** [comment]
Can you also update Merlin helm chart in caraml-dev/helm-charts: https://github.com/caraml-dev/helm-charts/tree/main/charts/merlin ?

**Comment by terryyylim** [comment]
> Can you also update Merlin helm chart in caraml-dev/helm-charts: https://github.com/caraml-dev/helm-charts/tree/main/charts/merlin ?

👌🏻 Yup! I'll update it once this chart has been tested with `0.26.0-rc4`.

**Pull request merged**